endpoints/openrtb2/auction.go

```diff
@@ -1132,28 +1132,34 @@ func validateRequestExt(req *openrtb_ext.RequestWrapper) []error {
 }
 
 func validateTargeting(t *openrtb_ext.ExtRequestTargeting) error {
-	if t != nil {
-		if t.PriceGranularity != nil {
-			if err := validatePriceGranularity(t.PriceGranularity); err != nil {
-				return err
-			}
+	if t == nil {
+		return nil
+	}
+
+	if t.PriceGranularity != nil {
+		if err := validatePriceGranularity(t.PriceGranularity); err != nil {
+			return err
 		}
-		if t.MediaTypePriceGranularity.Video != nil {
-			if err := validatePriceGranularity(t.MediaTypePriceGranularity.Video); err != nil {
+	}
+
+	if mediaPriceGranularity := t.MediaTypePriceGranularity; mediaPriceGranularity != nil {
```

Comment:
Why are you making a copy of `t.MediaTypePriceGranularity`? It doesn't seem to be to protect the original from accidental modification.

Reply:
This PR changes the MediaTypePriceGranularity field on the ExtRequestTargeting struct from a value to a pointer to match the rest of the targeting object. It looks like MediaTypePriceGranularity was intended to be a pointer since it has the omitemty directive which I noticed during this change wasn't working.

Therefore, this is not a copy. I'm assigning `t.MediaTypePriceGranularity` to a local variable to simplify the references from `t.MediaTypePriceGranularity` to directly `mediaPriceGranularity `.

Reply:
Changed for clarity. If this syntax isn't clear, likely not a great idea.